CMakeLists.txt

```diff
@@ -231,4 +253,16 @@ endif()
 
 if(CUKE_ENABLE_EXAMPLES)
     add_subdirectory(examples)
+
+    #check if c++11 is needed
+    if(${Qt5Core_FOUND})
+        if(NOT ${Qt5Core_VERSION_STRING} VERSION_LESS 5.7)
+            message(STATUS "C++11 is needed from Qt version 5.7.0, building with c++11 enabled")
+            set_target_properties(calcqt libcalcqt PROPERTIES
```

Comment:
Looking at the `Qt5CoreConfigExtras.cmake` file in [Qt 5.7.1 (as present on Debian stretch)](https://packages.debian.org/stretch/qtbase5-dev) this shouldn't be necessary because of this line:
```cmake
set_property(TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_FEATURES cxx_decltype)
```

That makes it a usage requirement (`INTERFACE_\w+` are properties describing usage requirements automatically applied by CMake) to always compile this target with a compiler supporting `decltype`, that means C++11 (or C++0x on some very old compilers, but if that's not enough then that entire compiler won't work anyway). So it should already automatically switch to using C++11 for every target that links to `Qt5::Core` (via `target_link_libraries`).

So please remove this entire block.